lib/gdpr/gateway/user_details.rb

```diff
@@ -29,4 +55,32 @@ def obfuscate_sponsors
         SET ud1.sponsor = REPLACE(ud1.sponsor, SUBSTRING_INDEX(ud1.sponsor, '@', '1'), 'user')
         WHERE ud2.username IS NULL")
   end
+
+  def notify_inactive_users
+    logger = Logger.new($stdout)
+    logger.info("Starting notification process for users inactive for 11 months")
+
+    inactive_users = DB[:userdetails].select(:username, :contact).where(
+      Sequel.lit("
+      (last_login < DATE_SUB(NOW(), INTERVAL 11 MONTH)
+      OR (last_login IS NULL AND created_at < DATE_SUB(NOW(), INTERVAL 11 MONTH)))
+      AND username != 'HEALTH'
+      "),
+    ).all
+
+    logger.info("Found #{inactive_users.size} inactive users")
+
+    inactive_users.each do |user|
+      contact = user[:contact]
+      username = user[:username]
+
+      if contact.start_with?("+")
```

Comment:
you could move this to the WifiUser model, so you could use `if contact.email?` so this logic doesn't need to be duplicated